akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -543,8 +552,8 @@ object Flow {
    *
    * '''Cancels when''' downstream cancels
    */
-  def lazyInit[I, O, M](flowFactory: I ⇒ Future[Flow[I, O, M]], fallback: () ⇒ M): Flow[I, O, M] =
-    Flow.fromGraph(new LazyFlow[I, O, M](flowFactory, fallback))
+  def lazyInitAsync[I, O, M](flowFactory: () ⇒ Future[Flow[I, O, M]]): Flow[I, O, Future[Option[M]]] =
```

Comment:
Perhaps we should deprecate/rename `Sink.lazyInit` also to make the naming consistent?

Reply:
I agree. If Sink.lazyInit is deprecated there should be a corresponding Sink.lazyInitAsync implementation.

@gosubpl  Do you want to take care of that in a separate pull request?

Reply:
@swachter @patriknw up to you, gentlemen 😉

Reply:
Go for it!

Reply:
Sadly, might not happen this month.

Reply:
Ok. Then I will extend this pull request to cover Sink.lazyInit, too.

Reply:
👍